### PR TITLE
[WIP/DO NOT MERGE] improve SVN poller & checkout step documentation

### PR DESCRIPTION
**Pull request by sa2ajj**
The seemingly common problem is: how do I track all branches in my SVN project and then build the detected changes?

The current examples are not complete and I spent some time trying to figure out how to do it after a user on IRC channel asked for help.  Another user said he uses old-style svn checkout step because it has `defaultBranch` option, which is not present in the new style step.

### CODE REVIEW

**Comment by sa2ajj** [comment]
The idea is create a "Guides" section where these SVN related things are covered (in this PR) and then populate the section with similar guides for other version control systems or other difficult to grasp concepts.

The first draft for SVN is available as http://bb.mawhrin.net/guides/svn.html

File: master/docs/guides/svn.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
+=================================
+Subversion Version Control System
```
Comment (djmitche) [inline master/docs/guides/svn.rst]: Maybe use a more detailed description for this particular guide?

Comment (sa2ajj) [inline master/docs/guides/svn.rst]: I'm open for suggestions :)

That was the best I could come out with.

Comment (djmitche) [inline master/docs/guides/svn.rst]: Maybe split each of the sections here into a separate guide?  Those after the first could refer back to the first ("You should be familiar with..").  So the guide heading would be "Building Every Commit to Trunk with Subversion"

Comment (sa2ajj) [inline master/docs/guides/svn.rst]: I considered at least two other guides:

* git (various practices specific for git related components)
* github (changes, status, maybe something else)

In the light of these ideas, a guide for SVN looked logical.  If I split all these (and following guides) into really small parts, then it's gonna be something else...

Comment (djmitche) [inline master/docs/guides/svn.rst]: That does seem like a lot of guides, it's true.  Maybe I just need to adjust the expectation I have from the word "guide" -- these are each focused on a particular usage topic that might involve content from all over the documentation (particularly for version-control systems, which will reference both change sources and source steps).

Comment (sa2ajj) [inline master/docs/guides/svn.rst]: > these are each focused on a particular usage topic that might involve content from all over the documentation (particularly for version-control systems, which will reference both change sources and source steps).

This is exactly what I have in mind.

Let's return to the topic of title when I think this "guide" is ready.

**Comment by djmitche** [comment]
I like "Cookbook", as you just suggested in irc.

**Comment by djmitche** [comment]
Should we keep this open?

**Comment by sa2ajj** [comment]
Yes, please do.  I hope to return to this RSN(tm).

**Comment by sa2ajj** [comment]
@djmitche on the second though, I close this (I still have the branch, do not I :))

**Pull request closed without merging**